src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,58 @@
-import React from 'react';
+/* eslint-disable react/require-default-props */
+import React, { useEffect, useState } from 'react';
+import { getUserPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+interface Props {
+  userId: string;
+  selectPost: (arg0?: Post) => void;
+  post?: Post;
+}
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+export const PostsList: React.FC<Props> = ({ userId, selectPost, post }) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
+  async function finder() {
+    const result = await getUserPosts(userId);
 
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+    setPostList(result);
+  }
+
+  useEffect(() => {
+    finder();
+  }, [userId]);
+
+  return (
+    <div className="PostsList">
+      <h2>Posts:</h2>
+      <ul className="PostsList__list">
+        {
+          currentPostList.map(singlePost => (
+            <li className="PostList__item">
+              <div>
+                <b>{`[User] #${singlePost.userId} `}</b>
+                {singlePost.title}
+              </div>
+              <button
+                type="button"
+                className="PostsList__button button"
+                onClick={() => {
+                  if ((post) && (post.id === singlePost.id)) {
+                    selectPost(undefined);
+                  } else {
+                    selectPost(singlePost);
+                  }
+                }}
```

Comment:
<img width="641" alt="image" src="https://user-images.githubusercontent.com/30314756/175943049-09a232c0-6f2b-45a3-ad9d-f2e7e607dd8c.png">
You should change selected post id, not full post